tests/rptest/tests/consumer_group_test.py

```diff
@@ -405,21 +406,29 @@ def test_group_recovery(self):
         assert offsets[TopicPartition(self.topic_spec.name,
                                       0)].leader_epoch > 0
 
+        # Remember the old offsets to compare them after the restart.
+        prev_offsets = offsets
+
         # Restart the broker.
         self.logger.info("Restarting redpanda nodes.")
         self.redpanda.restart_nodes(self.redpanda.nodes)
-        self.redpanda._admin.await_stable_leader("controller",
-                                                 partition=0,
-                                                 namespace='redpanda',
-                                                 timeout_s=60,
-                                                 backoff_s=2)
-
-        prev_offsets = offsets
 
         # Validate that the group state is recovered.
-        test_admin = KafkaTestAdminClient(self.redpanda)
-        offsets = test_admin.list_offsets(
-            group_id, [TopicPartition(self.topic_spec.name, 0)])
+        def list_offsets():
+            try:
+                test_admin = KafkaTestAdminClient(self.redpanda)
+                return test_admin.list_offsets(
+                    group_id, [TopicPartition(self.topic_spec.name, 0)])
+            except Exception as e:
+                self.logger.debug(f"Failed to list offsets: {e}")
+                return None
```

Comment:
> redpanda will reply with error_code::not_coordinator

I'm not sure I understand this, because this error is a normal error which a Kafka client should retry transparently. The original CI failure had this `assert len(offsets) == 1` failing, suggesting that `list_offsets` wasn't failing with an exception. I think I'm missing something here about why this fixes the issue.

Reply:
Sorry I still don't understand. Are there actually multiple logical changes in the same commit?

> The test didn't handle the top level error, now it does.

What error is this? Is it different than not_coordinator?

Reply:
Yes, it was 2 changes. Did split them now.
The error is `errro_code::not_coordinator`.